tests/test_l1dbproto.py

```diff
@@ -0,0 +1,475 @@
+
+        # Check the DIASources round trip properly. We don't need to be
+        # as complex here as the call signature has been almost fully tested
+        # here by the ``test_store_catalog_dia_sources`` tests.
+        self.assoc_db.store_dia_sources(dia_sources,
+                                        range(5),
+                                        self.exposure)
+        round_trip_dia_source_catalog = self.assoc_db.db.getDiaSources(
+            range(5), dt=None)
+
+        # Remake the DIASources with the correct values and columns for
+        # comparison.
+        dia_sources = create_test_points(
+            point_locs_deg=source_centers,
+            start_id=0,
+            schema=make_minimal_dia_source_schema(),
+            scatter_arcsec=-1,
+            associated_ids=range(5))
+        for dia_source in dia_sources:
+            dia_source['filterName'] = self.exposure.getFilter().getName()
+            dia_source['ccdVisitId'] = \
+                self.exposure.getInfo().getVisitInfo().getExposureId()
+            dia_source['psFlux'] = 10000. / self.flux0
+            dia_source['psFluxErr'] = np.sqrt(
+                (100. / self.flux0) ** 2 +
+                (10000. * self.flux0_err / self.flux0 ** 2) ** 2)
+
+        for stored_dia_source, dia_source in zip(round_trip_dia_source_catalog,
+                                                 dia_sources):
+
+            self._compare_source_records(stored_dia_source, dia_source)
+
+    def test_store_dia_sources(self):
+        """Test the storage of DIASources in the database.
+        """
+        # Create test associated DIAObjects and DIASources.
+        n_sources = 5
+        source_centers = [[1. * idx, 1. * idx] for idx in range(n_sources)]
+        obj_ids = [idx for idx in range(n_sources)]
+        dia_sources = create_test_points(
+            point_locs_deg=source_centers,
+            start_id=0,
+            schema=self.assoc_db.get_dia_source_schema(),
+            scatter_arcsec=1.0,
+            associated_ids=range(5))
+        for dia_source in dia_sources:
+            dia_source['psFlux'] = 10000. + np.random.randn() * 100.
+            dia_source['psFluxErr'] = 100. + np.random.randn() * 10.
+
+        # Check the DIASources round trip properly. We don't need to be
+        # as complex here as the call signature has been almost fully tested
+        # here by the ``test_store_catalog_dia_sources`` tests.
+        self.assoc_db.store_dia_sources(dia_sources,
+                                        range(5),
+                                        self.exposure)
+        round_trip_dia_source_catalog = self.assoc_db.db.getDiaSources(
+            range(5), dt=None)
+
+        for stored_dia_source, dia_source, obj_id in zip(
+                round_trip_dia_source_catalog,
+                dia_sources,
+                obj_ids):
+            dia_source['diaObjectId'] = obj_id
+            tmp_flux = dia_source['psFlux']
```

Comment:
There's some code duplication here with lines 408 - 414.

Reply:
I don't think it's duplicated. I create dia_sources that have extra columns that I don't want to write to the DB. The next set of code creates the correct DIASources as expected. Does that make sense?